loaders/src/glTF/errors.ts

```diff
@@ -0,0 +1,18 @@
+import { BaseError } from "babylonjs/Misc/baseError";
+
+/**
+ * Application error when unexpected magic number is found when loading file.
+ */
+export class GlTFLoaderUnexpectedMagicError extends BaseError {
```

Comment:
Do we really want such a specific error for this? Can it be more generic like `GLTFLoaderError`?

Also, small nit
```suggestion
export class GLTFLoaderUnexpectedMagicError extends BaseError {
```

Reply:
The intention here was to get the specific error, localize it and show to end user (of our app) so that they can self-recover. I believe this unexpected magic error is fixable by user.

Reply:
Hmm, okay. There are lots of error from the glTF loader that can be fixed by the user. Are we going to eventually add a custom error to each one? Maybe we can use an error code instead?

Reply:
Yes error codes make more sense. I believe we'd have to eventually officially document those. What is the guidance there? Check in the code first and followup with doc team?

Reply:
Ideally merge the code then follow up with an update to the docs. We can start with just one error code for now if this is all you need.